### DISCUSSION TITLE
Find predicted ID for entire cluster instead of each query cell

### DISCUSSION THREAD

**Discussion by danielcgingerich** [satijalab/seurat#4596] [Q&A]
I have two datasets, 1 scATAC and 1 scRNA.  Both come from the same tissue and same donor.   I would like to know what clusters in ATAC to compare to what clusters in RNA.

I understand how the prediction scores for each cell are found by multiplying L*t(W)
where L is the binary label matrix, W is the anchor weight matrix.  What this basically does is for each query cell, all of its anchors are partitioned by ident and their weights in each ident are summed.  Highest sum is the predicted id.  

If I wanted to do this for a cluster instead of individual query cells, could I simply take all anchors in the cluster, and sum the weights for each ident to find overall predicted cluster ID?

**Comment by timoast** [comment]
This should be equivalent to averaging the prediction score for each cell type within a cluster

**Reply by danielcgingerich** [reply]
@timoast that makes sense, I understand how that is the case.  

I tried this approach and it seems a bit bias towards the biggest clusters.  For example, my reference oligodendrocytes have 8 different clusters, my query oligodendrocytes have 5 different clusters.  Most query clusters end up mapping to the biggest reference oligodendrocyte cluster.  So I tried normalizing the anchor weights by dividing them by the size of the cluster that the reference cell belonged to, which helped some but still not satisfied with the results.

Since the cells from ATAC and RNA seq come from the same dissociated nuclei solution I have reason to believe the clusters should map to each other.  The last thing I want to try is separating the reference/query datasets by cell type, and then running CCA anchors between the reference/query.  I believe if only one cell type is included, then dim reduction will capture more variation within a single cell type than it would with all cell types combined.  This might result in better mapping.  Any thoughts?
